### ISSUE TITLE
Factorio

### ISSUE THREAD

**Issue by luka177** [skyline-emu/skyline-games-list#1255]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Tested Build

Edge-16

### Android Version

12

### SoC type

Snapdragon

### Device GSMArena link

https://www.gsmarena.com/google_pixel_3_xl-9257.php

### Game Behaviour

Game hangs at Factorio splash screen with strange analoug nice at right side near to factorio title. Tested on few devies both mtk and qcom with same result.
![Screenshot_20221207-081454_Skyline](https://user-images.githubusercontent.com/15714420/206113569-455c32f9-de7c-4b4e-9bf2-2fcc8290c312.png)

### Is Log Level is set to 'Info'?

- [X] Yes

### Log file

[emulation.sklog.zip](https://github.com/skyline-emu/skyline-games-list/files/10173641/emulation.sklog.zip)

**Comment by Gabrris** [comment]
absolutely same problems and can't wait to see them fixed. Entering the game, it shows this screen without a loading bar, that little glitch in the corner of the icon and played a little loading music, and thats all. Xiaomi11t (Mediatek, android12)

**Comment by Facitori** [comment]
Exactly the same thing here. Can't wait to play this game. My device is a samsung tab s8+ (SM-X800)

Already love the person who solved that problem

**Comment by luka177** [comment]
Same issue as #1139 ig, Partially supported RT format: X8B8G8R8 used!

**Comment by timas130** [comment]
I managed to get through the "Cropping bitmaps" step by copying `crop-cache.dat` from a working save on Ryujinx, but now I'm stuck on "Checking sprites" at 44%. Logs are similar to other comments in this issue.

![Screenshot_20230501-202833_Skyline](https://user-images.githubusercontent.com/20511387/235498372-396fe750-45ca-43ed-aa54-8e08fcafff49.png)

**Comment by dima-xd** [comment]
> I managed to get through the "Cropping bitmaps" step by copying `crop-cache.dat` from a working save on Ryujinx, but now I'm stuck on "Checking sprites" at 44%. Logs are similar to other comments in this issue.
> 
> ![Screenshot_20230501-202833_Skyline](https://user-images.githubusercontent.com/20511387/235498372-396fe750-45ca-43ed-aa54-8e08fcafff49.png)

can you please provide this save?

**Comment by timas130** [comment]
@dima-xd In-app save management doesn't export the cache, so you'll have to copy the file into `Android/data/skyline.emu/files/switch/nand/user/save/cache/01004200189F4000/crop-cache.dat` yourself.

Here's the file itself (packed into an archive so I can send it through GitHub): [crop-cache.dat.zip](https://github.com/skyline-emu/skyline-games-list/files/11412490/crop-cache.dat.zip)

Note that this is for Factorio 1.1.80. The file includes the game version as three 16-bit integers at the start, and the cache is ignored if versions don't match. Email me if you're having troubles with the ROM.

**Comment by dima-xd** [comment]
> @dima-xd In-app save management doesn't export the cache, so you'll have to copy the file into `Android/data/skyline.emu/files/switch/nand/user/save/cache/01004200189F4000/crop-cache.dat` yourself.
> 
> Here's the file itself (packed into an archive so I can send it through GitHub): [crop-cache.dat.zip](https://github.com/skyline-emu/skyline-games-list/files/11412490/crop-cache.dat.zip)
> 
> Note that this is for Factorio 1.1.80. The file includes the game version as three 16-bit integers at the start, and the cache is ignored if versions don't match. Email me if you're having troubles with the ROM.

is this the only one file from ryujinx's cache?

**Comment by timas130** [comment]
> is this the only one file from ryujinx's cache?

Yes, there are no more files except `config/config.ini`, which is in the "proper" save directory. It does include some settings related to graphics and stuff, though I haven't seen it in Skyline's save directory. Here it is:

[config.zip](https://github.com/skyline-emu/skyline-games-list/files/11412497/config.zip)

**Comment by dima-xd** [comment]
> > is this the only one file from ryujinx's cache?
> 
> Yes, there are no more files except `config/config.ini`, which is in the "proper" save directory. It does include some settings related to graphics and stuff, though I haven't seen it in Skyline's save directory. Here it is:
> 
> [config.zip](https://github.com/skyline-emu/skyline-games-list/files/11412497/config.zip)

thanks, I'll try to look into it to see why the game isn't working

**Comment by dima-xd** [comment]
> I managed to get through the "Cropping bitmaps" step by copying `crop-cache.dat` from a working save on Ryujinx, but now I'm stuck on "Checking sprites" at 44%. Logs are similar to other comments in this issue.
> 
> ![Screenshot_20230501-202833_Skyline](https://user-images.githubusercontent.com/20511387/235498372-396fe750-45ca-43ed-aa54-8e08fcafff49.png)

actually logs are not similar. The game crashes itself on 44%
`7:14:35.509  I  Service: ILogger::Log
17:14:35.511  I  factorio-run:MainThread:855703s: Factorio crashed. Generating symbolized stacktrace, please wait ...
17:14:35.511  I  Service: ILogger::Log
17:14:35.512  I  factorio-run:MainThread:855703s: Stack trace logging done
17:14:35.512  I  Service: ILogger::Log
17:14:35.512  I  factorio-run:MainThread:855703s:   37.226 Error DrawCommandBatch.cpp:409: Failed to get buffer for font 138 vertices.
17:14:35.513  I  Service: IApplicationFunctions::SetTerminateResult
17:14:35.513  I  SetTerminateResult: App set termination result: 674
17:14:35.526  E  Break: Exit Stack Trace (0)
                 * 0x79D8CC45EC (skyline::kernel::svc::Break(skyline::DeviceState const&) from libskyline.so)
                 * 0x79D8CA5500 (skyline::nce::NCE::SvcHandler(unsigned short, skyline::nce::ThreadContext*) from libskyline.so)
                 * 0x803C7A0CC (from sdk.nso.patch)
                 * 0x803D098B8 (from sdk.nso)
                 * 0x803D099E4 (from sdk.nso)
                 * 0x803D0A8C4 (from sdk.nso)
                 * 0x80423475C (from sdk.nso)
                 * 0x8041ED454 (from sdk.nso)
                 * 0x80128B310 (from main.nso)
                 * 0x801286B28 (from main.nso)
                 * 0x80063C22C (from main.nso)
                 * 0x8006F6354 (from main.nso)
                 * 0x800671A3C (from main.nso)
                 * 0x80017FB24 (from main.nso)
                 * 0x800673AF4 (from main.nso)
                 * 0x8006743C4 (from main.nso)
                 * 0x800680480 (from main.nso)
                 * 0x8005F37A0 (from main.nso)
                 * 0x8005FA320 (from main.nso)
                 * 0x8005FB584 (from main.nso)
                 * 0x800179C7C (from main.nso)
                 * 0x8001BFF40 (from main.nso)
                 * 0x800016A68 (main from main.nso)
                 * 0x800021BAC (nnMain from main.nso)
                 * 0x803D76414 (nn::init::Start(unsigned long, unsigned long, void (*)(), void (*)(), void (*)()) from sdk.nso)
                 * 0x8000010D4 (from rtld.nso)`

**Comment by timas130** [comment]
Oh, I actually thought that this is because I closed the game and these are the logs for that. Sorry!
